src/components/encounter/encounter-form.component.tsx

```diff
@@ -522,6 +522,25 @@ const EncounterForm: React.FC<EncounterFormProps> = ({
           isLowContrast: true,
         });
       }
+      // handle encounter diagnosis
+      try {
+        const saveDiagnoses = await EncounterFormManager.saveDiagnosis(fields, savedEncounter);
+        if (saveDiagnoses) {
+          showSnackbar({
+            title: t('encounterDiagnosisSaved', 'Encounter diagnosis saved successfully'),
+            kind: 'success',
+            isLowContrast: true,
+          });
+        }
```

Comment:
So the EncounterResource [supports diagnoses as a create-able property](https://github.com/openmrs/openmrs-module-webservices.rest/blob/fcc0ee7380d8e1ac861204d68e82d812f39bb1aa/omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/EncounterResource2_2.java#L44) shouldn't we be creating the diagnoses in a single transaction with the encounter?

Reply:
Let me add it the same way we handle orders